### PR TITLE
Fix #205: Fails CI when a don't merge label get added on a pull request.

### LINKED ISSUE #205: Add a GitHub Action for the "PR: don't merge…" labels
Couldn't we add a GitHub actions check that fails on the presence of the "PR: don't merge…" labels & make it a required check?

I think we can listen for label changes, too, so that particular workflow re-runs if we did that, I'm pretty sure auto-merge would respect it since it has to match normal merging criteria (which for us is approvals + all required tests passing).

### PR DESCRIPTION
**Pull request by jameesjohn**
## Explanation
Fixes #205: Fails CI when a don't merge label gets added on a pull request. 

This PR has been tested at https://github.com/jameesjohn/comment-on-pr/pull/51

## Checklist
- [ ] I have successfully deployed my own instance of Oppiabot.
  - You can find instructions for doing this [here](https://github.com/oppia/oppiabot/wiki/Deploying-your-own-instance-of-the-oppiabot).
- [ ] I have manually tested all the changes made in this PR following the [manual tests matrix](https://github.com/oppia/oppiabot/wiki/Manual-Tests-Matrix).

### CODE REVIEW

File: constants.js [hunk 1]
```diff
@@ -17,8 +17,8 @@ const openEvent = 'opened';
 const openEventGithubActions = 'pull_request_target_opened';
 const reopenEventGithubActions = 'pull_request_target_reopened';
 const reopenEvent = 'pull_request_reopened';
-const unlabelEvent = 'unlabeled';
-const PRLabelEvent = 'labeled';
+const unlabelEvent = 'pull_request_unlabeled';
```
Comment (vojtechjelinek) [inline constants.js:20]:
```suggestion
const PRUnlabelEvent = 'pull_request_unlabeled';
```

File: actions/src/dispatcher.js [hunk 2]
```diff
@@ -40,6 +41,12 @@ module.exports = {
             case constants.claCheckGithubAction:
               await claCheckGithubActionModule.claCheckGithubAction();
               break;
+            case constants.prLabelCheck:
+              await pRLabelsModule.checkLabels();
```
Comment (vojtechjelinek) [inline actions/src/dispatcher.js:45]:
```suggestion
              await PRLabelsModule.checkLabelAdded();
```

File: actions/src/dispatcher.js [hunk 3]
```diff
@@ -40,6 +41,12 @@ module.exports = {
...
+              break;
+            case constants.dontMergeLabelCheck:
+              await pRLabelsModule.checkUnLabeled();
```
Comment (vojtechjelinek) [inline actions/src/dispatcher.js:48]:
```suggestion
              await PRLabelsModule.checkLabelRemoved();
```

File: actions/src/pull_requests/labelCheck.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
```
Comment (vojtechjelinek) [inline actions/src/pull_requests/labelCheck.js:1]: Should we make sure that the label cannot be removed by a person and only can be removed once there aren't any merge conflicts?

**Review by vojtechjelinek — CHANGES_REQUESTED** [review]
Thanks! Left a few comments.

File: actions/src/pull_requests/labelCheck.js (hunk 4 above)
Comment (jameesjohn) [reply actions/src/pull_requests/labelCheck.js:1]: There are multiple don't merge labels. Should this be done only for the `PR don't merge: merge conflict` label?

File: actions/src/dispatcher.js (hunk 2 above)
Comment (jameesjohn) [reply actions/src/dispatcher.js]: Done.

File: actions/src/dispatcher.js (hunk 3 above)
Comment (jameesjohn) [reply actions/src/dispatcher.js]: Done.

File: constants.js (hunk 1 above)
Comment (jameesjohn) [reply constants.js]: Done.

**Comment by vojtechjelinek** [comment]
@jameesjohn Is this ready for review?

File: actions/src/pull_requests/labelCheck.js (hunk 4 above)
Comment (vojtechjelinek) [reply actions/src/pull_requests/labelCheck.js:1]: Hmm, not sure, we can probably do it in a different PR thought.

**Comment by jameesjohn** [comment]
Yes @vojtechjelinek.

File: actions/src/pull_requests/labelCheck.js (hunk 4 above)
Comment (jameesjohn) [reply actions/src/pull_requests/labelCheck.js:1]: Yes.

**Review by vojtechjelinek — APPROVED** [review]
LGTM!

**Pull request merged**